paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -234,6 +236,12 @@ internal abstract class BaseSheetViewModel(
             isLinkEnabled = linkHandler.isLinkEnabled.asLiveData(),
             initialSelection = savedSelection.asLiveData(),
             isNotPaymentFlow = this is PaymentOptionsViewModel,
+            nameProvider = { code ->
+                val paymentMethod = lpmResourceRepository.getRepository().fromCode(code)
+                paymentMethod?.displayNameResource?.let {
+                    application.getString(it)
+                }.orEmpty()
+            }
```

Comment:
Not my proudest moment. I can follow-up and check if we can replace `displayNameResource` with `displayName` on `SupportedPaymentMethod`.